kectl/app/cmd/util/cloudinstaller.go

```diff
@@ -0,0 +1,249 @@
+package util
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"time"
+)
+
+type KubeCloudInstTool struct {
+	Common
+}
+
+func (cu *KubeCloudInstTool) InstallTools() error {
+	cu.SetOSInterface(GetOSInterface())
+	cu.SetKubeEdgeVersion(cu.ToolVersion)
+
+	err := cu.InstallKubeEdge()
+	if err != nil {
+
+		return err
+	}
+	fmt.Println("Installation of kubeedge package is sucessfull")
+	err = cu.generateCertificates()
+	if err != nil {
+		fmt.Println(" in err")
+		return err
+	}
+	fmt.Println("Certificates got generated at : /etc/kubeedge/kubeedge")
+	err = cu.tarCertificates()
+	if err != nil {
+		return err
+	}
+	fmt.Println("Certificates got Tared at : /etc/kubeedge/kubeedge/certificates folder, Please copy the certificates to the respective edge node")
+
+	err = cu.startKubernetescluster()
+	if err != nil {
+		return err
+	}
+	fmt.Println("Kubernetes cluster started")
+
+	err = cu.updateManifests()
+	if err != nil {
+		return err
+	}
+	fmt.Println("Updation of manifests is sucessful")
+
+	err = cu.updateControlleryaml()
+	if err != nil {
+		return err
+	}
+	fmt.Println("Updation of controller yaml is sucess")
+
+	time.Sleep(10 * time.Second)
```

Comment:
please remove sleep and wait till apiserver exposes port 8080. Give a timeout time that if after 10 or 15 seconds the port is still not exposed, return error